de.metas.business/src/test/java/org/adempiere/impexp/bpartner/BPartnerImportProcess_MultiLocations_gh2543_Test.java

```diff
@@ -0,0 +1,255 @@
+import org.compiere.util.Env;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * Test case described in https://github.com/metasfresh/metasfresh/issues/2543
+ *
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+public class BPartnerImportProcess_MultiLocations_gh2543_Test
+{
+	private Properties ctx;
+
+	@Before
+	public void init()
+	{
+		AdempiereTestHelper.get().init();
+		ctx = Env.getCtx();
+	}
+
+	@Test
+	public void testMultiplePartnerAndAddresses()
+	{
+		final List<I_I_BPartner> ibpartners = prepareImportMultipleBPartners();
+		Assert.assertNotNull("list null", ibpartners);
+
+		final BPartnerImportProcess importProcess = new BPartnerImportProcess();
+		importProcess.setCtx(ctx);
+
+		final IMutable<Object> state = new Mutable<>();
+
+		ibpartners.forEach(importRecord -> {
+			try
+			{
+				importProcess.importRecord(state, importRecord);
+			}
+			catch (final Exception e)
+			{
+				throw AdempiereException.wrapIfNeeded(e); // propagate
+			}
+		});
+
+		assertMultipleBpartnerImported(ibpartners);
+
+	}
+
+	private void assertMultipleBpartnerImported(final List<I_I_BPartner> ibpartners)
+	{
+		ibpartners.forEach(BPartnerImportTestHelper::assertImported);
+
+		// check first partner imported
+		// should have 2 contacts and one address
+		assertFirstImportedBpartner(ibpartners);
+
+		// check second partner imported
+		// should have 1 contact and one address (the address is exactly the one from the first bpartner)
+		{
+			assertSecondImportedBpartner(ibpartners);
+
+			// check location - is similar with the one from first partner, but should have different id
+			final I_C_BPartner firstBpartner = ibpartners.get(0).getC_BPartner();
+			final I_C_BPartner secondBPartner = ibpartners.get(2).getC_BPartner();
+			final List<de.metas.adempiere.model.I_C_BPartner_Location> fbplocations = Services.get(IBPartnerDAO.class).retrieveBPartnerLocations(firstBpartner);
+			final List<de.metas.adempiere.model.I_C_BPartner_Location> sbplocations = Services.get(IBPartnerDAO.class).retrieveBPartnerLocations(secondBPartner);
+			Assert.assertTrue(fbplocations.get(0).getC_Location_ID() != sbplocations.get(0).getC_Location_ID());
```

Comment:
I this, using assertj this would be easier to read: `assertThat(fbplocations.get(0).getC_Location_ID()).isNotEqualTo(sbplocations.get(0).getC_Location_ID())`

Reply:
agree. i missed that while refactoring